tests/v2/actions/charts/stackstate.go

```diff
@@ -285,3 +362,31 @@ func newStackstateAgentChartUpgradeAction(p *payloadOpts, stackstateConfigs *obs
 
 	return chartUpgradeAction
 }
+
+// CreateClusterRepo creates a new ClusterRepo resource in the Kubernetes cluster using the provided catalog client.
+// It takes the client, repository name, and repository URL as arguments and returns an error if the operation fails.
+func CreateClusterRepo(client *rancher.Client, catalogClient *catalog.Client, name, url string) error {
+	ctx := context.Background()
+	repo := buildClusterRepo(name, url)
+	_, err := catalogClient.ClusterRepos().Create(ctx, repo, metav1.CreateOptions{})
+
+	client.Session.RegisterCleanupFunc(func() error {
+
+		var propagation = metav1.DeletePropagationForeground
+		err := catalogClient.ClusterRepos().Delete(context.Background(), name, metav1.DeleteOptions{PropagationPolicy: &propagation})
+		if err != nil {
+			return err
+		}
+
+		return err
+	})
+	return err
+}
+
+// buildClusterRepo creates and returns a new ClusterRepo object with the provided name and URL.
+func buildClusterRepo(name, url string) *rv1.ClusterRepo {
+	return &rv1.ClusterRepo{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Spec:       rv1.RepoSpec{URL: url},
+	}
+}
```

Comment:
We probably don't need a separate function for this instantiation as it is only used once anyway.

Reply:
I was just trying to adhere to the single responsibility principle.

Reply:
it goes back to having too much abstraction, we can define it in the calling function..

Reply:
yeah SRP makes sense, but I think in this case since it's only used the one time it should just part of the calling function. If we foresee this being extended or used more then perhaps it makes sense to be its own function. I'll let you make that call.

Reply:
sounds like you want to keep this, which is fine. Plus a lot of this will unfortunately be stripped out once StackState team has their own automation built, so LGTM here.